src/details/ArborX_NeighborList.hpp

```diff
@@ -39,8 +39,7 @@ struct NeighborListPredicateGetter
 
     auto const &hyper_point =
         reinterpret_cast<::ArborX::Point<dim, Coordinate> const &>(pair.value);
```

Comment:
Do you still need the `reinterpret_cast`?

Reply:
Could be replaced by `bit_cast` but the issue is same as above: user data.